### PR TITLE
numpydoc set_size_inches docstring

### PR DESCRIPTION
**Pull request by dstansby**
n.b. I have no idea what the description for the `forward` argument really means, so if anyone has a better suggestion than what is present please let me know!

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -876,18 +876,24 @@ def figimage(self, X, xo=0, yo=0, alpha=None, norm=None, cmap=None,
...
-        ACCEPTS: a (w, h) tuple with w, h in inches
+        Parameters
+        ----------
+        w : tuple of floats or float
+            Width and height in inches (if height not specified as a separate
+            argument) or width.
+        h : float
+            Height in inches.
+        forward : bool
+            If ``True``, the cavas size is automatically updated, e.g., you can
```
Comment (anntzer) [inline lib/matplotlib/figure.py]: typo: canvas

**Comment by tacaswell** [comment]
If forward is `False` then the figure and the GUI window can get out of sync which means you will either see just part of the figure or the figure won't take up the whole GUI.  We need this knob so that when we re-size the Figure from the GUI we don't get our selves into an infinite loop!

File: lib/matplotlib/figure.py (hunk 1 above)
Comment (tacaswell) [inline lib/matplotlib/figure.py]:
```suggestion
            If ``True``, the canvas size is automatically updated, e.g., you can
```

File: lib/matplotlib/figure.py [hunk 2]
```diff
@@ -876,18 +876,24 @@ def figimage(self, X, xo=0, yo=0, alpha=None, norm=None, cmap=None,
...
-        automatically updated; e.g., you can resize the figure window
-        from the shell
-
-        ACCEPTS: a (w, h) tuple with w, h in inches
+        Parameters
+        ----------
+        w : tuple of floats or float
+            Width and height in inches (if height not specified as a separate
+            argument) or width.
+        h : float
```
Comment (tacaswell) [inline lib/matplotlib/figure.py]:
```suggestion
        h : float, optional
```

Specifying the default here does not make lots of sense due to the weird interaction with `w`.

File: lib/matplotlib/figure.py [hunk 3]
```diff
@@ -876,18 +876,24 @@ def figimage(self, X, xo=0, yo=0, alpha=None, norm=None, cmap=None,
...
+            Height in inches.
+        forward : bool
```
Comment (tacaswell) [inline lib/matplotlib/figure.py]:
```suggestion
        forward : bool, default: True
```

File: lib/matplotlib/figure.py [hunk 4]
```diff
@@ -876,18 +876,24 @@ def figimage(self, X, xo=0, yo=0, alpha=None, norm=None, cmap=None,
...
              fig.set_size_inches((w, h))
 
-        optional kwarg *forward=True* will cause the canvas size to be
-        automatically updated; e.g., you can resize the figure window
-        from the shell
-
-        ACCEPTS: a (w, h) tuple with w, h in inches
+        Parameters
+        ----------
+        w : tuple of floats or float
```
Comment (timhoffm) [inline lib/matplotlib/figure.py]:
```suggestion
        w : (float, float) or float
```

**Comment by timhoffm** [comment]
flake8:

> ./lib/matplotlib/figure.py:895:80: E501 line too long (80 > 79 characters)

**Pull request merged**